File: programming/Leetcode/Container_With_Most_Water_11.c
int maxArea(int* height, int heightSize){
    
    int max = 0, left = 0, right = heightSize - 1;
    
    while(left < right)
    {
        int width = right - left;
        int high = 0;
        
        if (height[left] < height[right])
        {
            high = height[left];
            left++;
        }
        else
        {
            high = height[right];
            right--;
        }
        
        int tmp = width*high;
        
        if (tmp > max)
        {
            max = tmp;
        }
    }
    
    return max;
        
}

File: programming/Leetcode/Container_With_Most_Water_11.py
# time complexity : O(n)
# space complexity : O(1)

class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        water = 0
        
        while right != left:
            
            water = max(min(height[left], height[right])*(right-left), water)
            
            if height[left] > height[right]:
                right = right - 1
            else:
                left = left + 1
                
        return water
